pyinfra/api/connectors/winrm.py

```diff
@@ -0,0 +1,213 @@
+from __future__ import print_function, unicode_literals
+
+import click
+import winrm
+
+from pyinfra import logger
+from pyinfra.api import Config
+from pyinfra.api.exceptions import ConnectError
+from pyinfra.api.util import memoize
+
+from .util import make_win_command
+
+
+def _raise_connect_error(host, message, data):
+    message = '{0} ({1})'.format(message, data)
+    raise ConnectError(message)
+
+
+@memoize
+def show_warning():
+    logger.warning('The @winrm connector is alpha!')
+
+
+def _make_winrm_kwargs(state, host):
+    kwargs = {
+    }
+
+    for key, value in (
+        ('username', host.data.winrm_user),
+        ('password', host.data.winrm_password),
+        ('winrm_port', int(host.data.winrm_port or 0)),
+        ('timeout', state.config.CONNECT_TIMEOUT),
+    ):
+        if value:
+            kwargs[key] = value
+
+    # Password auth (boo!)
+    if host.data.winrm_password:
+        kwargs['password'] = host.data.winrm_password
+
+    # TODO: add more auth
+    # pywinrm supports: basic, certificate, ntlm, kerberos, plaintext, ssl, credssp
+    # see https://github.com/diyan/pywinrm/blob/master/winrm/__init__.py#L12
+
+    return kwargs
+
+
+def make_names_data(hostname):
+
+    show_warning()
+
+    yield '@winrm/{0}'.format(hostname), {'winrm_hostname': hostname}, []
+
+
+def connect(state, host):
+    '''
+    Connect to a single host. Returns the winrm Session if successful.
+    '''
+
+    kwargs = _make_winrm_kwargs(state, host)
+    logger.debug('Connecting to: {0} ({1})'.format(host.name, kwargs))
+
+    # TODO: create a ~/.winrm/config file?
```

Comment:
I think we can ignore this as it's out of pyinfra scope - I only mentioned in case it was a known thing but if not, no problem :)

Reply:
Ok